canonicalwebteam/blog/common_view_logic.py

```diff
@@ -20,46 +26,48 @@ def get_index_context(page_param, articles, total_pages):
 
         category_ids = article["categories"]
 
+        # Can these calls be bundled?
+        first_item = True
         for category_id in category_ids:
             if category_id not in category_cache:
-                category_cache[category_id] = {}
-
+                try:
+                    resolved_category = api.get_category_by_id(category_id)
+                except Exception:
```

Comment:
It's not good practice to catch a generic `Exception`. If we must catch exceptions, we should be as explicit about it as we can be, so we're not unintentionally masking other errors. (Although I can see you simply copied this from how it used to be, I think we should take this opportunity to improve it).

Why would this category_id not exist anyway? That seems like it shouldn't happen - and therefore should raise an error?

Reply:
The main culprit that makes this necessary is the following `process_response` function in `wordpress_api`:
```
def process_response(response):
    if not response.ok:
        raise Exception("Error from api: " + response.status_code)

    return response.json()
```
Since this will throw on anything but 200 we have to catch the error somehow. A simple timeout would otherwise kill the server.

What do you think of returning `None` as the default behaviour when encountering an error?

Reply:
Maybe we should just do `response.raise_for_status` and catching the native requests error?

Reply:
Have updated it accordingly. Thanks